specifyweb/frontend/js_src/lib/components/DataModel/saveBlockers.tsx

```diff
@@ -131,12 +154,34 @@ export function getFieldBlockers(
   resource: SpecifyResource<AnySchema>,
   field: LiteralField | Relationship
 ): RA<string> {
-  const blockers = saveBlockers.get(resource)?.blockers ?? [];
+  const blockers = getResourceBlockers(resource).blockers;
   return blockers
     .filter((blocker) => blocker.field === field)
     .map(({ message }) => message);
 }
 
+function usePreviouslySetBlocker(
+  resource: SpecifyResource<AnySchema>,
+  field: LiteralField | Relationship
+): GetSet<FieldValue> {
+  const fieldAndValue = {
+    [field.name]: resource.get(field.name),
+  };
+  if (!previouslySetBlockers.has(resource))
```

Comment:
how is `previouslySetBlockers` different from `getResourceBlockers`?
i.e current save blockers are saved somewhere already, correct? so can we get them from there or does that not work?

Reply:
The difference is that `previouslySetBlockers` stores the previous value of the field the last time the resource and field was called with `setSaveBlockers`. 
The resource blockers only store the message of the blocker which by itself is not enough information to determine whether it is safe to reset the blockers for the field if whichever consumer of the function is trying to clear the blockers. 

The rationale behind `previouslySetBlockers` is that we know we should NOT reset any existing save blockers if the value for the field has not changed since the last time  `setSaveBlockers` has been called for that field. 
(i.e., we should only allow the blockers to be reset if the field value has changed since the last call to `setSaveBlockers`)